projects/kubernetes-sigs/cluster-api/patches/0002-Add-unstacked-etcd-support.patch

```diff
@@ -539,109 +540,17 @@ index fde450016..943898620 100644
 +	return endpoints, found, nil
 +}
 diff --git a/controlplane/kubeadm/api/v1beta1/kubeadm_control_plane_webhook.go b/controlplane/kubeadm/api/v1beta1/kubeadm_control_plane_webhook.go
-index 896ef5d0e..9d5e58664 100644
+index c20d90e1d..424139bc3 100644
 --- a/controlplane/kubeadm/api/v1beta1/kubeadm_control_plane_webhook.go
 +++ b/controlplane/kubeadm/api/v1beta1/kubeadm_control_plane_webhook.go
-@@ -133,6 +133,7 @@ func (in *KubeadmControlPlane) ValidateUpdate(old runtime.Object) error {
+@@ -137,6 +137,7 @@ func (in *KubeadmControlPlane) ValidateUpdate(old runtime.Object) error {
  		{spec, kubeadmConfigSpec, clusterConfiguration, "etcd", "local", "imageRepository"},
  		{spec, kubeadmConfigSpec, clusterConfiguration, "etcd", "local", "imageTag"},
  		{spec, kubeadmConfigSpec, clusterConfiguration, "etcd", "local", "extraArgs", "*"},
 +		{spec, kubeadmConfigSpec, clusterConfiguration, "etcd", "external", "endpoints"},
  		{spec, kubeadmConfigSpec, clusterConfiguration, "dns", "imageRepository"},
  		{spec, kubeadmConfigSpec, clusterConfiguration, "dns", "imageTag"},
  		{spec, kubeadmConfigSpec, clusterConfiguration, "imageRepository"},
-diff --git a/controlplane/kubeadm/config/crd/bases/controlplane.cluster.x-k8s.io_kubeadmcontrolplanes.yaml b/controlplane/kubeadm/config/crd/bases/controlplane.cluster.x-k8s.io_kubeadmcontrolplanes.yaml
-index 8adfcd777..03da30b18 100644
---- a/controlplane/kubeadm/config/crd/bases/controlplane.cluster.x-k8s.io_kubeadmcontrolplanes.yaml
-+++ b/controlplane/kubeadm/config/crd/bases/controlplane.cluster.x-k8s.io_kubeadmcontrolplanes.yaml
-@@ -2177,6 +2177,20 @@ spec:
-                               upgrades.
-                             type: string
-                         type: object
-+                      proxy:
-+                        description: Proxy holds the https and no proxy information
-+                          This is only for bottlerocket
-+                        properties:
-+                          httpsProxy:
-+                            description: HTTPS proxy
-+                            type: string
-+                          noProxy:
-+                            description: No proxy, list of ips that should not use
-+                              proxy
-+                            items:
-+                              type: string
-+                            type: array
-+                        type: object
-                     type: object
-                   mounts:
-                     description: Mounts specifies a list of mount points to be setup.
-@@ -2878,6 +2892,20 @@ spec:
-                               upgrades.
-                             type: string
-                         type: object
-+                      proxy:
```

Comment:
So we are just moving this to the first patch instead?

Reply:
yeah, for a couple of patches that changed API structs, the `generate-manifests` hadn't been run to update CRDs until the next patch. But this time I ran the generate command as soon as i applied the patch to see if there were any CRD changes. This caused CRD changes to moved from one patch to another, in this case, the previous one.

Reply:
This is also why I removed patch 11 - adding generated manifests. That change had already gone into a preceding patch.